pkg/function/create_csi_snapshot_static_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	. "gopkg.in/check.v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	dynfake "k8s.io/client-go/dynamic/fake"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/kanisterio/kanister/pkg/kube/snapshot"
+)
+
+type CreateCSISnapshotStaticTestSuite struct {
+	snapshotName   string
+	namespace      string
+	deletionPolicy string
+	driver         string
+	snapshotHandle string
+	snapshotClass  string
+}
+
+var _ = Suite(&CreateCSISnapshotStaticTestSuite{})
+
+func (testSuite *CreateCSISnapshotStaticTestSuite) SetUpSuite(c *C) {
+	testSuite.snapshotName = "test-snapshot"
+	testSuite.namespace = "test-namespace"
+	testSuite.deletionPolicy = "Retain"
+	testSuite.driver = "test-driver"
+	testSuite.snapshotClass = "test-snapshot-class"
+	testSuite.snapshotHandle = "test-snapshot-handle"
+}
+
+func (testSuite *CreateCSISnapshotStaticTestSuite) TestCreateCSISnapshotStatic(c *C) {
+	for _, api := range []*metav1.APIResourceList{
```

Comment:
I prefer not to promote them to globals because the existing `CreateCSISnapshotTestSuite` already has some overlapping global variables like snapshot name, namespace, snapshot class etc. I think it's ok to not couple these tests with a small amount of test code repetition.